src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscoveryManager.cs

```diff
@@ -123,11 +123,9 @@ public void DiscoverTests(DiscoveryCriteria discoveryCriteria, ITestDiscoveryEve
             // If there are sources to discover
             if (verifiedExtensionSourceMap.Any())
             {
-                new DiscovererEnumerator(_requestData, discoveryResultCache, _cancellationTokenSource.Token).LoadTests(
-                    verifiedExtensionSourceMap,
-                    RunSettingsUtilities.CreateAndInitializeRunSettings(discoveryCriteria.RunSettings),
-                    discoveryCriteria.TestCaseFilter,
-                    _sessionMessageLogger);
+                var runSettings = RunSettingsUtilities.CreateAndInitializeRunSettings(discoveryCriteria.RunSettings);
```

Comment:
This did a lot in the the method call. This way it is much easier to see what is passed on.